File: fund-management-backend/app/api/routes/loans.py
# Loans endpoints placeholder
from fastapi import APIRouter, Depends, status

from app.core.responses import success_response
from app.dependencies.permissions import require_admin
from app.schemas.loan_schema import CreateLoanRequest
from app.services.loan_service import LoanService

router = APIRouter(
    prefix="/funds/{fund_id}/loans",
    tags=["Loans"],
)


@router.post(
    "",
    status_code=status.HTTP_201_CREATED,
)
async def create_loan(
    fund_id: str,
    request: CreateLoanRequest,
    current_user=Depends(require_admin),
):

    loan = await LoanService.create_loan(
        fund_id=fund_id,
        member_id=request.member_id,
        loan_amount=request.loan_amount,
        created_by=current_user["id"],
    )

    return success_response(
        message="Loan created successfully.",
        data=loan,
    )


@router.get("")
async def get_all_loans(
    fund_id: str,
    current_user=Depends(require_admin),
):

    loans = await LoanService.get_all_loans(
        fund_id=fund_id,
    )

    return success_response(
        message="Loans fetched successfully.",
        data=loans,
    )


@router.get("/active")
async def get_active_loans(
    fund_id: str,
    current_user=Depends(require_admin),
):

    loans = await LoanService.get_active_loans(
        fund_id=fund_id,
    )

    return success_response(
        message="Active loans fetched successfully.",
        data=loans,
    )


@router.get("/member/{member_id}")
async def get_member_loans(
    fund_id: str,
    member_id: str,
    current_user=Depends(require_admin),
):

    loans = await LoanService.get_member_loans(
        fund_id=fund_id,
        member_id=member_id,
    )

    return success_response(
        message="Member loans fetched successfully.",
        data=loans,
    )


@router.get("/{loan_id}")
async def get_loan(
    fund_id: str,
    loan_id: str,
    current_user=Depends(require_admin),
):

    loan = await LoanService.get_loan(
        loan_id=loan_id,
    )

    return success_response(
        message="Loan fetched successfully.",
        data=loan,
    )

File: fund-frontend/services/loans.js
import { api } from "@/lib/api";

// POST /funds/{fundId}/loans { member_id, loan_amount }  (admin only)
export async function createLoan(fundId, payload) {
  return api.post(`/funds/${fundId}/loans`, payload);
}

// GET /funds/{fundId}/loans
export async function getAllLoans(fundId) {
  return api.get(`/funds/${fundId}/loans`);
}

// GET /funds/{fundId}/loans/active
export async function getActiveLoans(fundId) {
  return api.get(`/funds/${fundId}/loans/active`);
}

// GET /funds/{fundId}/loans/member/{memberId}
export async function getMemberLoans(fundId, memberId) {
  return api.get(`/funds/${fundId}/loans/member/${memberId}`);
}

// GET /funds/{fundId}/loans/{loanId}
export async function getLoan(fundId, loanId) {
  return api.get(`/funds/${fundId}/loans/${loanId}`);
}
